src/main/java/com/google/sps/servlets/CreateGameServlet.java

```diff
@@ -113,7 +113,7 @@ private int getNumStages(HttpServletRequest request) {
     *         the jth hint of the ith stage.
```

Comment:
As for the "CreateGameServlet.java", those methods are correlated with each other. For example, the returning value of List<Integer> getNumHints should match the number of elements for each row in List<List<Coordinates>> getHintLocations(). What would happen if the returning values of both methods don't agree with each other. It is aways better to have a single-source-of-truth. For example, we might have some methods like below. I didn't put too much thoughts when listing the following methods. Please use your judgment when create those methods. 

private int getNumStages();

private int getNumHints(int stageIndex);

private List<Coordinates> getHintLocations(int stageIndex);

private Coordinate getHintLocation(int stageIndex, int hintIndex);

Reply:
There's not much I can do here in regards to determining the single-source truth values, as these lists come directly from a parameters in a POST request. I've instead added a checker that makes sure the lists have sizes that agree, and will print an exception and terminate if not.

Reply:
Why do you need to use the "HttpServletRequest request" as the argument for all private methods?

Reply:
The request contains all of the fetch parameters I need to parse information about the newly created game.

Reply:
Use a centralized method to parse the HttpServletRequest. So it is easy to check the JSON structure of the request and extend the JSON structure in the future..

Reply:
I have already listed out the full JSON structure needed in the Java API doc. The link to that doc can be found in the appendix of our design doc.